app/uk/gov/hmrc/customs/declarations/information/model/actionbuilders/actionBuildersModel.scala

```diff
@@ -87,6 +91,12 @@ case class ConversationIdRequest[A](conversationId: ConversationId,
                                     request: Request[A]
 ) extends WrappedRequest[A](request) with HasRequest[A] with HasConversationId
 
+// Available after ShutterCheckAction
+case class ApiVersionRequest[A](conversationId: ConversationId,
+                                requestedApiVersion: ApiVersion,
+                                request: Request[A]
+) extends WrappedRequest[A](request) with HasRequest[A] with HasConversationId with HasApiVersion
+
 // Available after ValidatedHeadersAction builder
 case class ValidatedHeadersRequest[A](conversationId: ConversationId,
```

Comment:
ValidatedHeadersRequest and AuthorisedRequest should have HasApiVersion trait. Might affect logging.